src/users/users.service.ts

```diff
@@ -58,9 +63,105 @@ export class UsersService {
     return this.usersRepository.update(id, updateUserDto);
   }
 
+  updateByUsername(
+    username: string,
+    updateUserDto: UpdateUserDto,
+  ): Promise<UserModel> {
+    const message = `UsersService.updateByUsername() username=${username} updateUserDto=${JSON.stringify(
+      updateUserDto,
+    )}`;
+    this.logger.log(message);
+    return this.usersRepository.updateByUsername(username, updateUserDto);
+  }
```

Comment:
same reason with `UsersController.updateByUsername`. maybe we can avoid adding this. let me know if it doesn't.